sql/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -231,6 +231,14 @@ private RoutedCollectPhase createPhase(PlannerContext plannerContext) {
         );
     }
 
+    @Override
+    public LogicalPlan tryOptimize(@Nullable LogicalPlan pushDown) {
+        if (pushDown instanceof Order) {
+            return ((Order) pushDown).newInstance(this);
```

Comment:
Would it make sense to rename `newInstance` to `withNewChild` or `replaceChild` or something like that?

It's fairly clear what newInstance does if you look at the parameter name `newSource`, but on the call-site that's not visible.

Reply:
Yep. Good point.